control-plane/plugin/src/resources/volume.rs

```diff
@@ -236,3 +255,45 @@ impl CreateRows for HashMap<String, openapi::models::ReplicaTopology> {
             .collect()
     }
 }
+
+impl GetHeaderRow for openapi::models::RebuildHistory {
+    fn get_header_row(&self) -> Row {
+        (*utils::REBUILD_HISTORY_HEADER).clone()
+    }
+}
+
+impl CreateRows for openapi::models::RebuildHistory {
+    fn create_rows(&self) -> Vec<Row> {
+        self.records
+            .iter()
+            .map(|rec| {
+                row![
+                    rec.child_uri[get_index(rec.child_uri.clone()) ..],
+                    rec.src_uri[get_index(rec.src_uri.clone()) ..],
+                    rec.rebuild_job_state.to_string(),
+                    to_human_readable(rec.blocks_total),
+                    to_human_readable(rec.blocks_recovered),
+                    to_human_readable(rec.blocks_transferred),
+                    to_human_readable(rec.blocks_remaining),
+                    to_human_readable(rec.block_size),
+                    rec.is_partial.to_string(),
+                    rec.start_time,
+                    rec.end_time,
+                ]
+            })
+            .collect()
+    }
+}
+
+fn get_index(val: String) -> usize {
+    for (i, v) in val.char_indices() {
+        if v.to_string().as_str() == "=" {
+            return i + 1;
+        }
+    }
+    0_usize
+}
```

Comment:
Something like this:

```suggestion
fn child_uuid(uri: &str) -> String {
    let Ok(uri) = url::Url::from_str(uri) else { return "".into(); } ;
    match uri.query_pairs().find(|(q, _)| q == "uuid") {
      Some((_, uuid)) => uuid.to_string(),
      _ => "".into()
    }
}
```